lib/utils/systemd/systemd.go

```
@@ -0,0 +1,85 @@
+package systemd
+
+import (
+	"context"
+	"os/exec"
+
+	"github.com/coreos/go-systemd/v22/dbus"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+// Systemctl runs a local systemctl command in non-blocking mode.
+// TODO(knisbet): I'm using systemctl here, because using go-systemd and dbus appears to be unreliable, with
+// masking unit files not working. Ideally, this will use dbus at some point in the future.
+func Systemctl(ctx context.Context, operation, service string) error {
+	return SystemctlCmd(ctx, operation, service, "--no-block")
+}
+
+// SystemctlCmd executes the command for systemctl
+func SystemctlCmd(ctx context.Context, operation, service string, args ...string) error {
+	args = append([]string{operation, service}, args...)
+	out, err := exec.CommandContext(ctx, "/bin/systemctl", args...).CombinedOutput()
+	log.WithFields(log.Fields{
+		"operation": operation,
+		"output":    string(out),
+		"service":   service,
+	}).Info("Execute systemctl.")
+	if err != nil {
+		return trace.Wrap(err, "failed to execute systemctl: %s", out).AddFields(map[string]interface{}{
+			"operation": operation,
+			"service":   service,
+		})
+	}
+	return nil
+}
+
+// TryResetService will request for systemd to restart a system service
+func TryResetService(ctx context.Context, service string) {
+	// ignoring error results is intentional
+	err := Systemctl(ctx, "restart", service)
+	if err != nil {
+		log.Warn("error attempting to restart service", err)
```

nit: logging at `Warn` and ignoring the error don't complement each other IMO. If the restart is important, I'd return an error, if not (e.g. if service is expected to exhibit transient start up failures and will be restarted by systemd to success) - I'd log it at `Debug` as otherwise it will be spamming the logs.